test/crawler_web/controllers/api/v1/user_registration_controller_test.exs

```diff
@@ -0,0 +1,84 @@
+defmodule CrawlerWeb.Api.V1.UserRegistrationControllerTest do
+  use CrawlerWeb.ConnCase, async: true
+
+  alias Crawler.Account.Guardian
+
+  describe "POST create/2" do
+    test "given valid user credentials, returns access token", %{conn: conn} do
+      email = "new_email@gmail.com"
+
+      conn =
+        post(conn, Routes.api_v1_user_registration_path(conn, :create), %{
+          email: email,
+          password: "12345678"
+        })
+
+      assert %{
+               "data" => %{
+                 "attributes" => %{
+                   "email" => ^email,
+                   "token" => _
+                 },
+                 "id" => _,
+                 "relationships" => %{},
+                 "type" => "user_registration"
+               },
+               "included" => []
+             } = json_response(conn, 200)
+    end
+
+    test "given invalid user credentials, returns error response", %{conn: conn} do
+      conn = post(conn, Routes.api_v1_user_registration_path(conn, :create), %{})
+
+      assert json_response(conn, 422) == %{
+               "errors" => [
+                 %{
+                   "code" => "unprocessable_entity",
+                   "detail" => gettext("Email or password is missing")
+                 }
+               ]
+             }
+    end
+
+    test "given existing user credentials, returns error response", %{conn: conn} do
+      %{email: email, password: password} = insert(:user)
+
+      conn =
+        post(conn, Routes.api_v1_user_registration_path(conn, :create), %{
+          email: email,
+          password: password
+        })
+
+      assert json_response(conn, 422) == %{
+               "errors" => [
+                 %{
+                   "code" => "unprocessable_entity",
+                   "detail" => gettext("%{email} has already been taken", email: email)
+                 }
+               ]
+             }
+    end
+
+    test "given valid user credentials when encode_and_sign returns error, returns error response",
```

Comment:
what happen if we `call this endpoint as a logged in user` @vnntsu, should we have that test case

Reply:
@andyduong1920 I haven't handled the case yet. An authorized user can request registration API. Should we handle it or could we leave it now?

Reply:
1. You can add a test for existing behavior
2. We might not need to handle it now (if we don't have time)
3. You can create a separate issue for that, and we could work on it later on

But a test for the existing behavior is a must to have ;)

Reply:
@andyduong1920 Added in [67ea950](https://github.com/vnntsu/elixir-ic/pull/45/commits/67ea950a96fa2d96ea57ea0d375a2efa47ebebb8)